src/test/java/StringManipulationTest.java

```diff
@@ -20,65 +22,104 @@ public void tearDown() {
     public void testRemoveNthCharacter1() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
         assertEquals("I' bttr uts0e 16tsinths trn6 rgh?", manipulatedstring.removeNthCharacter(3, false));
     }
 
     @Test
+    // Checks behavior for preserving whitespace
     public void testRemoveNthCharacter2() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
         assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
     }
 
     @Test
+    // Checks behavior for preserving white space with n = 1 (all characters removed)
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I <3 strings");
+        assertEquals("            ", manipulatedstring.removeNthCharacter(1, true));
     }
 
     @Test
+    // Checks behavior for not preserving white space with n = 1 (all characters removed)
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I <3 strings");
+        assertEquals(null, manipulatedstring.removeNthCharacter(1, false));
     }
 
     @Test
+    // Checks behavior for removing interval of characters greater than string length
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I <3 strings");
+        Executable executable = new Executable() {
+            public void execute() {
+                manipulatedstring.removeNthCharacter(20, false);
+            }
+        };
+        IndexOutOfBoundsException exception = assertThrows(IndexOutOfBoundsException.class, executable);
+        assertEquals("n cannot be greater than the length of the string", exception.getMessage());
     }
 
     @Test
+    // Checks behavior for removing interval of characters <0
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I <3 strings");
+        Executable executable = new Executable() {
+            public void execute() {
+                manipulatedstring.removeNthCharacter(-1, false);
+            }
+        };
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, executable);
+        assertEquals("n must be greater than 0", exception.getMessage());
     }
 
     @Test
+    // Checks behavior for removing characters near end of string
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I <3 strings");
+        assertEquals("I <3 string ", manipulatedstring.removeNthCharacter(12, true));
     }
 
     @Test
+    // Checks behavior for substrings
     public void testGeSubStrings1() {
         manipulatedstring.setString("This is my string");
         String [] sStings = manipulatedstring.getSubStrings(3, 4);
```

Comment:
//don't forget at the end
assertEquals("Expected result", expectedArray, sStrings);